gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -472,25 +514,22 @@ func joinPath(paths ...string) string {
 	return strings.Join(paths, "/")
 }
 
-// dirname returns the logical parent directory of the path.  This is different
-// than path.Split() in that we want dirname("foo/bar/") -> "foo/", whereas
-// path.Split() returns "foo/bar/".
-func dirname(path string) string {
-	leading := ""
-	if strings.HasPrefix(path, "/") {
-		leading = "/"
-	}
+// getParent returns the logical parent directory of the path.  This is different
+// than path.Split() in that we want getParent("/gcs/foo/bar/") -> "/gcs/foo/", whereas
+// path.Split() returns "/gcs/foo/bar/".
+// As a special case, getParent returns the empty string for the bucket root, e.g.
+// getParent("/gcs/foo") -> "".
+func getParent(path string) string {
```

Comment:
Probably not a hard requirement to change, but we've got `filepath.Dir()` in the stdlib with well-understood behavior, might be a more appropriate thing to do than roll our own `getParent()`

Reply:
True, simplified this function as well in a new commit.